src/components/organisms/NavigatorPane/ResourceNavigator.tsx

```diff
@@ -0,0 +1,102 @@
+import {useRef} from 'react';
+
+import {Skeleton} from 'antd';
+
+import styled from 'styled-components';
+
+import {useAppSelector} from '@redux/hooks';
+import {resourceNavigatorSelector} from '@redux/selectors/resourceSelectors';
+
+import {elementScroll, useVirtualizer} from '@tanstack/react-virtual';
+
+import EmptyResourceNavigator from './EmptyResourceNavigator';
+import KindRenderer from './KindRenderer';
+import ResourceRenderer from './ResourceRenderer';
+import {useScroll} from './useScroll';
+
+function ResourceNavigator() {
+  const list = useAppSelector(resourceNavigatorSelector);
+  const ref = useRef<HTMLUListElement>(null);
+  const isLoading = useAppSelector(
+    state =>
+      state.ui.isFolderLoading || state.main.previewOptions.isLoading || state.main.clusterConnectionOptions.isLoading
+  );
+
+  const rowVirtualizer = useVirtualizer({
+    count: list.length,
+    estimateSize: () => 28,
```

Comment:
Why do we estimate this to 28?

Reply:
That's the height of each row in the navigator, I'll create a constant for the number